app/frontend/schools/courses/types/students_editor/StudentsEditor__CSVDataError.res

```diff
@@ -1,10 +1,16 @@
 type errorType =
   | Name
+  | InvalidName
   | Email
+  | InvalidEmail
   | Title
+  | InvalidTitle
   | TeamName
+  | InvalidTeamName
   | Tags
+  | InvalidTags
   | Affiliation
+  | InvalidAffiliation
```

Comment:
It looks like we have a lot of code duplication, why don't we add a new variant `HasInvalidCharacters(array<string>)`

The array of stings can be the keys of the columns that could be used to render errors at each location.

Reply:
I'm not very familiar with rescript, please explain the usage of the new variant suggested regarding the existing system (type errorType)

Reply:
@bodhish hi sir, can you give me an example of the usage of `HasInvalidCharacters(array<string>)`?

Reply:
I would make it something like this; 

```res
type errorVariants =
  | InvalidFormat
  | InvalidCharacters

type errorType =
  | Name
  | Email
  | Title
  | TeamName
  | Tags
  | Affiliation

type error = {
  errorType: errorType,
  variant: errorVariants,
}

type t = {
  rowNumber: int,
  errors: array<error>,
}
```

Reply:
thanks @bodhish for the explanation